pkg/rke2/psp/psp.go

```diff
@@ -0,0 +1,443 @@
+package psp
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/rancher/spur/cli"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/api/policy/v1beta1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	yamlutil "k8s.io/apimachinery/pkg/util/yaml"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/transport"
+)
+
+const (
+	defaultRetries          = 5
+	defaultWaitSeconds      = 5
+	defaultTimeout          = 30
+	k8sWrapTransportTimeout = 30
+)
+
+const (
+	kubeSystemNamespace = "kube-system"
+	cisAnnotationValue  = "resolved"
+
+	namespaceAnnotationBase               = "psp.rke2.io/"
+	namespaceAnnotationGlobalRestricted   = namespaceAnnotationBase + "global-restricted"
+	namespaceAnnotationGlobalUnrestricted = namespaceAnnotationBase + "global-unrestricted"
+	namespaceAnnotationSystemUnrestricted = namespaceAnnotationBase + "system-unrestricted"
+)
+
+// setGlobalUnrestricted sets the global unrestricted podSecurityPolicy with
+// the associated role and rolebinding.
+func setGlobalUnrestricted(cs *kubernetes.Clientset, ns *v1.Namespace) error {
+	if _, ok := ns.Annotations[namespaceAnnotationGlobalUnrestricted]; !ok {
+		if _, err := cs.RbacV1().ClusterRoles().Get(context.TODO(), globalUnrestrictedPSPName, metav1.GetOptions{}); err != nil {
```

Comment:
`context.TODO()`
are we not setting up a proper signal handling context and passing it around in rke2?
Id expect something like this to exist: https://github.com/rancher/k3s/blob/master/pkg/cli/server/server.go#L219

Looks like this PR is the first to need to create a context, so i think it unfortunately falls on you to pull in  a proper  signalhandling context

Reply:
This appears to be a no go. k3s uses it and it's implementation panics if it's called twice so calling it before we get to k3s isn't going to work.